pkg/config/xds/xds.go

```diff
@@ -71,20 +77,92 @@ func BuildXDSObjectFromStruct(applyTo networking.EnvoyFilter_ApplyTo, value *str
 	return obj, nil
 }
 
-func StructToMessage(pbst *structpb.Struct, out proto.Message, strict bool) error {
-	if pbst == nil {
+type anyTransformer struct {
+	unknownTypes sets.String
+}
+
+var typedStructTypeURL = protoconv.TypeURL(&udpa.TypedStruct{})
+
+func (a *anyTransformer) AnyToTypedStruct(pbst *structpb.Struct) *structpb.Struct {
```

Comment:
Can you please add some tests for this?

Reply:
Is there any possibility that it panics?

Reply:
Added tests. I don't see any possibility of a panic